src/services/sync/strategies/LDAPSyncer.js

```diff
@@ -32,111 +39,152 @@ class LDAPSyncer extends Syncer {
-								return this.app
-									.service('schools')
-									.update({ _id: schools.data[0]._id }, { $set: { name: ldapSchool.displayName } });
-							}
-							return this.app
-								.service('schools')
-								.update({ _id: schools.data[0]._id }, { $set: { name: schools.data[0].name } });
-						}
-
-						return this.getCurrentYearAndFederalState().then(({ currentYear, federalState }) => {
-							const schoolData = {
-								name: ldapSchool.displayName,
-								systems: [this.system._id],
-								ldapSchoolIdentifier: ldapSchool.ldapOu,
-								currentYear,
-								federalState,
-							};
-							newSchools += 1;
-							return this.app.service('schools').create(schoolData);
-						});
-					})
-					.catch((err) => {
-						fails += 1;
-						this.logError(`School creation failed for ${ldapSchool.displayName}`, err);
-					})
-			)
-		).then((res) => {
-			this.logInfo(`Created ${newSchools} new schools and updated ${updates} schools. ${fails} schools failed`);
-			return Promise.resolve(res);
-		});
+					};
+					this.syncQueue.sendToQueue(schoolData, {});
+					schoolList.push(schoolData.data);
+				} catch (err) {
+					this.logger.error('Uncaught LDAP sync error', { error: err, systemId: this.system._id, syncId: this.syncId });
+				}
+			}
+		} catch (err) {
+			this.logger.error('Uncaught LDAP sync error', { error: err, systemId: this.system._id, syncId: this.syncId });
+		}
+		return schoolList;
+	}
+
+	async getUserData(school) {
+		this.logInfo(`Getting users for school ${school.name}`, { syncId: this.syncId });
+		const ldapUsers = await this.app.service('ldap').getUsers(this.system.ldapConfig, school);
+		this.logInfo(`Creating and updating ${ldapUsers.length} users for school ${school.name}`, { syncId: this.syncId });
+
+		const bulkSize = 1000; // 5000 is a hard limit because of definition in user model
+
+		for (let i = 0; i < ldapUsers.length; i += bulkSize) {
+			const ldapUserChunk = ldapUsers.slice(i, i + bulkSize);
+			for (const ldapUser of ldapUserChunk) {
+				try {
+					const userData = {
+						action: LDAP_SYNC_ACTIONS.SYNC_USER,
+						syncId: this.syncId,
+						data: {
+							user: {
+								firstName: ldapUser.firstName,
+								lastName: ldapUser.lastName,
+								systemId: this.system._id,
+								schoolDn: school.ldapSchoolIdentifier,
+								email: ldapUser.email,
+								ldapDn: ldapUser.ldapDn,
+								ldapId: ldapUser.ldapUUID,
+								roles: ldapUser.roles,
+							},
+							account: {
+								ldapDn: ldapUser.ldapDn,
+								ldapId: ldapUser.ldapUUID,
+								username: `${school.ldapSchoolIdentifier}/${ldapUser.ldapUID}`.toLowerCase(),
+								systemId: this.system._id,
+								schoolDn: school.ldapSchoolIdentifier,
+								activated: true,
+							},
+						},
+					};
+					this.syncQueue.sendToQueue(userData, {});
+				} catch (err) {
+					this.logError(`User creation error for ${ldapUser.firstName} ${ldapUser.lastName} (${ldapUser.email})`, {
```

Comment:
The error message seems to be a little bit confusing to me. The code in the try block is only about sending the message to the queue, isn't it?